submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/StatsMongoDataService.kt

```diff
@@ -1,62 +1,76 @@
+import ac.uk.ebi.biostd.persistence.doc.db.repositories.SubmissionMongoRepository
+import ac.uk.ebi.biostd.persistence.doc.model.DocSubmissionStats
 import ac.uk.ebi.biostd.persistence.doc.model.SingleSubmissionStat
 import org.springframework.data.domain.PageRequest
 
 class StatsMongoDataService(
-    private val statsDataRepository: SubmissionStatsDataRepository
+    private val submissionsRepository: SubmissionMongoRepository,
+    private val statsDataRepository: SubmissionStatsDataRepository,
 ) : StatsDataService {
+    override fun findByAccNo(accNo: String): List<SubmissionStat> =
+        statsDataRepository
+            .findByAccNo(accNo)
+            ?.stats
+            ?.map { SingleSubmissionStat(accNo, it.value, SubmissionStatType.valueOf(it.key)) }
+            ?: throw StatsNotFoundException(accNo)
+
     override fun findByType(submissionStatType: SubmissionStatType, filter: PaginationFilter): List<SubmissionStat> =
         statsDataRepository
             .findAllByStatType(submissionStatType, PageRequest.of(filter.pageNumber, filter.limit))
             .content
-            .map { toSubmissionStat(it, submissionStatType) }
+            .map { toSubmissionStat(submissionStatType, it) }
 
     override fun findByAccNoAndType(accNo: String, submissionStatType: SubmissionStatType): SubmissionStat =
         statsDataRepository
             .findByAccNoAndStatType(accNo, submissionStatType)
-            ?.let { toSubmissionStat(it, submissionStatType) }
+            ?.let { toSubmissionStat(submissionStatType, it) }
             ?: throw StatNotFoundException(accNo, submissionStatType)
 
-    // TODO support saving stats for expired versions Pivotal ID # 176802128
+    override fun save(stat: SubmissionStat): SubmissionStat {
+        updateOrRegister(stat)
+        return stat
+    }
+
     override fun saveAll(stats: List<SubmissionStat>): List<SubmissionStat> =
         stats.filterValid()
             .mapValues { it.value.last() }
-            .map { updateSubmissionStat(it.key!!, it.value) }
+            .map { updateOrRegister(it.value) }
+
 
     override fun incrementAll(stats: List<SubmissionStat>): List<SubmissionStat> =
         stats.filterValid()
-            .mapValues { incrementSubmissionStat(it.key!!, it.value) }
+            .mapValues { increment(it.key, it.value) }
             .values
             .toList()
 
     private fun List<SubmissionStat>.filterValid() =
         groupBy { it.accNo.uppercase() }
-            .mapKeys { statsDataRepository.findByAccNo(it.key.uppercase()) }
-            .filterKeys { it != null }
+            .filter { submissionsRepository.existsByAccNo(it.key) }
 
-    private fun updateSubmissionStat(sub: DocSubmission, update: SubmissionStat): SubmissionStat {
-        statsDataRepository.updateStat(sub.accNo, sub.version, update)
+    private fun updateOrRegister(update: SubmissionStat): SubmissionStat {
+        statsDataRepository.updateOrRegisterStat(update)
```

Comment:
rather that return error, we should return repository value so api always return what is database.